pkg/networkserver/grpc_gsns.go

```diff
@@ -835,7 +835,7 @@ func (ns *NetworkServer) handleJoin(ctx context.Context, up *ttnpb.UplinkMessage
 	}
 
 	req := &ttnpb.JoinRequest{
-		CFList:             frequencyplans.CFList(*fp, dev.LoRaWANPHYVersion),
+		CFList:             frequencyplans.TrimmedCFList(*fp, dev.LoRaWANPHYVersion),
```

Comment:
I think you should;
1. Make sure that `resetMACState()` only selects the first five frequencies and puts those in `CurrentParameters`. Leave that function also responsible
2. Use the `CurrentParameters` here in a call to get the `CFList` based on that

Reply:
I don't agree with @johanstokking's second point. The `CurrentParameters` are a **result** of the CFList, so they shouldn't be input to the function that calculates the CFList.

Why do we need a separate `frequencyplans.TrimmedCFList` func? I don't think there is any "frequencies" CFList than one with max 5 entries, so why not just add this to the normal `CFList` func?

Reply:
My point is that the magic of selecting _N_ and _which_ N frequencies are part of CFList should be in one place. If we put that in `CFList`, fine, but then we should add an argument to `resetMACState()` saying which are current.

Note that we are talking about the current parameters of the _pending_ session. So that already assumes it's pending, not that it is already current. It will be current once the device confirms the session. So I stand by my point 2.

Reply:
I'm totally fine with applying CFList to the CurrentParameters of the PendingSession. I'm just saying that these CurrentParameters shouldn't be an input to the CFList calculation function.